src/test/java/com/google/step/youtube/DescriptionParserTest.java

```diff
@@ -34,140 +35,181 @@ public void parse_matchesFromNone() {
+        OfferSnippet expectedPEWDIEPIE = OfferSnippet.create("PEWDIEPIE", expectedSnippetPEWDIEPIE);
+        String expectedSnippetCOOLIRPA = "Save 33% on your first Native Deodorant Pack - normally "
+                + "$36, you'll get it for $24! Click here: https://bit.ly/nativecoolirpa and use "
+                + "my code \"COOLIRPA\".";
+        OfferSnippet expectedCOOLIRPA = OfferSnippet.create("COOLIRPA", expectedSnippetCOOLIRPA);
+        String expectedSnippetNordVPN = "Go to https://NordVPN.com/pewdiepie and use code "
+                + "PEWDIEPIE to get 70% off a 3 year plan plus 1 additional month free.";
+        OfferSnippet expectedNordVPN = OfferSnippet.create("https://NordVPN.com/pewdiepie", expectedSnippetNordVPN);
+        List<OfferSnippet> expected = 
+            Arrays.asList(expectedFUNGBROS10, expectedPEWDIEPIE, expectedCOOLIRPA, expectedNordVPN);
+        
+        assertThat(DescriptionParser.parse(desc), equalTo(expected));
     }
 
     @Test
     public void parse_codeNoQuotes() {
         String desc = "Get 10% off (save up to $44!) your own authentic Japanese snack box from "
-                + "Bokksu using my link: https://bit.ly/3fYbkZ5 and code FUNGBROS10";
-        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList("FUNGBROS10")));
+                + "Bokksu using my link: https://bit.ly/3fYbkZ5 and code FUNGBROS10"
+                + "\n\nRyan: https://www.instagram.com/ryan.w.benson/";
+        String expectedSnippet = "Get 10% off (save up to $44!) your own authentic Japanese snack "
+                + "box from Bokksu using my link: https://bit.ly/3fYbkZ5 and code FUNGBROS10";
+        OfferSnippet expected = OfferSnippet.create("FUNGBROS10", expectedSnippet);
+        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList(expected)));
     }
 
     @Test
     public void parse_codeWithQuotes() {
         String desc = "Save 33% on your first Native Deodorant Pack - normally $36, you'll get "
                 + "it for $24! Click here: https://bit.ly/nativecoolirpa and use my code \"COOLIRPA\". ";
-        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList("COOLIRPA")));
+        String expectedSnippet = "Save 33% on your first Native Deodorant Pack - normally $36, you'll get"
+        + " it for $24! Click here: https://bit.ly/nativecoolirpa and use my code \"COOLIRPA\". ";
+        OfferSnippet expected = OfferSnippet.create("COOLIRPA", expectedSnippet);
+        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList(expected)));
     }
 
     @Test
     public void parse_toAtLink() {
-        String desc = "Go to https://NordVPN.com/pewdiepie and"
-                + "get 70% off a 3 year plan plus 1 additional month free.";
-        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList("https://NordVPN.com/pewdiepie")));
+        String desc = "ty for 10 years of pewdiepie youtube uploads brvus\n\nGo to "
```

Comment:
Maybe you can more easily signify that 'ty for 10 years of pewdiepie youtube uploads brvus' is not in the snippet.
"DESCRIPTION \n\n Go to...."

Reply:
or "NOT_IN_SNIPPET \n\n Go to..."

Reply:
Factored out the actual snippets.